docs/sphinx/source/changelog/v2.1.0.rst

```diff
@@ -0,0 +1,90 @@
+  to be fit as soiling intervals based on  ``min_interval_length``,
+  ``max_relative_slope_error``, and ``max_negative_step``. (:pull:`199`, :issue:`272`)
+* The default ``day_scale`` parameter in soiling functions and methods was changed
+  from 14 to 13. A recommendation to use an odd value along with a warning for even
+  values was also added. (:pull:`199`, :issue:`189`)
+* The default ``min_interval_length`` in soiling functions and methods was changed
+  from 2 to 7. (:pull:`199`)
+
+Enhancements
+------------
+* New object oriented end-to-end analysis API introduced with the
+  py:class:`~rdtools.analysis_chains.TrendAnalysis` class. Note the behavior has
+  been adjusted since its original introduction in v2.1.0-beta.0. (:pull:`117`,
+  :pull:`232`, :pull:`233`, :pull:`263`, :pull:`278`, :pull:`285`, :pull:`281`)
+* Add ``sensor_filter_components`` and ``clearsky_filter_components`` to
+  :py:class:`~rdtools.analysis_chains.TrendAnalysis` (:issue:`236`, :pull:`263`)
+* A new parameter ``outlier_factor`` was added to soiling functions and methods to
+  enable better control of cleaning event detection. (:pull:`199`)
+* Boolean input kwarg ``detailed`` has been added to
+  :py:class:`~rdtools.plotting.degradation_summary_plots` to color-code 
+  degradation plots by the number of times data points are used in the
+  degradation distribution. (:issue:`269`, :pull:`282`)
+* :py:class:`~rdtools.degradation.degradation_year_on_year` adds a new 
+  ``usage_of_points`` entry in the ``calc_info`` return dictionary.
+  (:issue:`269`, :pull:`282`)
+* :py:func:`~rdtools.filtering.clip_filter` updated to allow for different methods of
+   clipping detection with the ``model`` parameter (:pull:`200`)
+* Add new function :py:func:`~rdtools.filtering.quantile_clip_filter` (:pull:`200`).
+* Add new function :py:func:`~rdtools.filtering.logic_clip_filter` (:pull:`200`).
+* Add new function :py:func:`~rdtools.filtering.xgboost_clip_filter` (:pull:`200`).
+* Add new function :py:func:`~rdtools.plotting.tune_filter_plot` (:pull:`200`).
+
+
+Bug fixes
+---------
+* Unexpected recoveries when using ``method=random_clean`` in the soiling module
+  have been fixed. (:pull:`199`, :issue:`234`)
+* Improved NaN pixel handling in
+  :py:func:`~rdtools.clearsky_temperature.get_clearsky_tamb` (:pull:`274`).
+
+
+
+Documentation
+-------------
+* Corrected a typo in the :py:class:`~rdtools.analysis_chains.TrendAnalysis`
+  docstring (:pull:`264`)
+* Enabled intersphinx so that function parameter types are linked to external
+  documentation (:pull:`258`)
+
+
+Requirements
+------------
+* Update pinned versions of several dependencies (:pull:`261`, :pull:`275`):
+
+    * ``requirements.txt``: cached-property, certifi, chardet, idna, matplotlib, numpy, Pillow,
+      requests, urllib
+    * ``docs/notebook_requirements.txt``: argon2-cffi, bleach, cffi, colorama, Jinja2,
+      numexpr, packaging, pycparser, pygments
```

Comment:
I think these lists are out of date to some extent (e.g. tornado, traitlets)

Reply:
Good point. I'll update to be more vague and just reference this PR. In reality this environment specifications aren't that critical and someone who is really in the weeds on some problem can just directly compare versions to figure out what has changed